File: api/sentiment/reddit/[query].js
export default function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  const { query } = req.query;
  const now = new Date();

  const posts = [
    {
      id: "abc123",
      title: `Discussion about ${query}'s latest trades`,
      subreddit: "wallstreetbets",
      author: "trader_joe",
      score: 1542,
      num_comments: 234,
      url: "https://reddit.com/r/wallstreetbets/comments/abc123",
      created_at: new Date(now - 2 * 60 * 60 * 1000).toISOString(),
      sentiment: "positive"
    },
    {
      id: "def456",
      title: `What do you think about ${query}?`,
      subreddit: "stocks",
      author: "market_watcher",
      score: 856,
      num_comments: 156,
      url: "https://reddit.com/r/stocks/comments/def456",
      created_at: new Date(now - 5 * 60 * 60 * 1000).toISOString(),
      sentiment: "neutral"
    },
    {
      id: "ghi789",
      title: `Breaking: ${query} makes major move`,
      subreddit: "investing",
      author: "finance_guru",
      score: 2103,
      num_comments: 412,
      url: "https://reddit.com/r/investing/comments/ghi789",
      created_at: new Date(now - 8 * 60 * 60 * 1000).toISOString(),
      sentiment: "positive"
    }
  ];

  res.json(posts);
}

File: api/sentiment/reddit/[query].py
from http.server import BaseHTTPRequestHandler
import json
import urllib.parse
from datetime import datetime, timedelta


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Extract query from path
        path_parts = self.path.split('/')
        query = urllib.parse.unquote(path_parts[-1].split('?')[0])

        # Mock Reddit posts
        posts = [
            {
                "id": "abc123",
                "title": f"Discussion about {query}'s latest trades",
                "subreddit": "wallstreetbets",
                "author": "trader_joe",
                "score": 1542,
                "num_comments": 234,
                "url": "https://reddit.com/r/wallstreetbets/comments/abc123",
                "created_at": (datetime.now() - timedelta(hours=2)).isoformat(),
                "sentiment": "positive"
            },
            {
                "id": "def456",
                "title": f"What do you think about {query}?",
                "subreddit": "stocks",
                "author": "market_watcher",
                "score": 856,
                "num_comments": 156,
                "url": "https://reddit.com/r/stocks/comments/def456",
                "created_at": (datetime.now() - timedelta(hours=5)).isoformat(),
                "sentiment": "neutral"
            },
            {
                "id": "ghi789",
                "title": f"Breaking: {query} makes major move",
                "subreddit": "investing",
                "author": "finance_guru",
                "score": 2103,
                "num_comments": 412,
                "url": "https://reddit.com/r/investing/comments/ghi789",
                "created_at": (datetime.now() - timedelta(hours=8)).isoformat(),
                "sentiment": "positive"
            }
        ]

        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        self.wfile.write(json.dumps(posts).encode())

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
